AutoResizeFont/DynamicLabel.swift

```diff
@@ -10,38 +10,60 @@ import UIKit
 
 @IBDesignable
 final class DynamicLabel: UILabel {
+    
+    //allow font to accept size from interface builder and scale up from there.
+    //this is to let us use values from the design specs
     @IBInspectable
-    var fontSizePts: CGFloat {
-        didSet {
-            calculateFontSize()
+    var pointSize = UIFont() {
+        didSet{
+            self.font = pointSize
+        }
+    }
+    
+    //override font for view
+    override var font : UIFont?{
+        didSet{
+            setNeedsLayout()
             setNeedsDisplay()
         }
     }
     
+    //deduced by multiplying the height of the Label by a constant
+    var relativeHeight : CGFloat!
+    
+    //relative constant to multiply our initial element height.
+    //this lets us double the size on bigger devices
+    let relativeFontConstant:CGFloat = 0.1
+    
     override init(frame: CGRect) {
-        fontSizePts = 14
-        super.init(frame: frame);
+        super.init(frame: frame)
+        relativeHeight = CGFloat(frame.height * relativeFontConstant)
+        calculateFontSize()
     }
     
     required init?(coder aDecoder: NSCoder) {
-        fontSizePts = 14
-        super.init(coder: aDecoder);
+        super.init(coder: aDecoder)
+        relativeHeight = CGFloat(frame.height * relativeFontConstant)
+        calculateFontSize()
     }
     
-    // todo calculate fonts for iPad and iPod
+    //calculate fonts for iPad, iPhone and iPod
     func calculateFontSize() {
```

Comment:
`calculateFontSize` is only called when the view is initialised. If the font size is changed subsequently, this calculation is not triggered.

Reply:
I am only calculating and applying the font properties on load of the view. are you referring to changes that may be made on the device due to accessibility settings?. Please clarify.

Reply:
No, changes that could be made while designing on IB or changes that could be effected from code after the app is already running.

Reply:
fixed in commit #618c4f7